dashboard-widgets.el

```diff
@@ -710,6 +710,22 @@ String -> bool.
 Argument IMAGE-PATH path to the image."
   (eq 'gif (image-type image-path)))
 
+
+(defun dashboard--is-image-animated-p (image-path)
+  "Return t if animated and has multiple frames, nil otherwise.
+   String -> bool.
+   Argument IMAGE-PATH: Path to the image.
+   This function first checks if the image type supports animation 
+   like GIF or WebP, and then verifies if the supported image has 
+   multiple frames."
+  (when (or (eq (image-type image-path) 'gif)
+            (eq (image-type image-path) 'webp))
+    (let ((image (create-image image-path)))
+      (if (image-multi-frame-p image)
+          t
+        nil))))
```

Comment:
There is no reason to return `nil`. If you return nothing, it will default to `nil`. Or even better, return the condition directly:

```elisp
  (when (or (eq (image-type image-path) 'gif)
            (eq (image-type image-path) 'webp))
    (let ((image (create-image image-path)))
      (image-multi-frame-p image)))
```

Reply:
I also though of this one initially. But I did this one just to return boolean(nil or t). The function image-multi-frame-p will return   a cons (nimages . delay), where nimages is the number of frames and delay is the delay in seconds between them. I think cons (nimages . delay) is considered true. So your fix  must be good to go with.